src/main/js/utils/generate.js

```diff
@@ -6,57 +6,109 @@ import {
   filteredCourses,
   startRedirect,
 } from '../actions/generator';
+import addGPA from './grade';
 import ScheduleMaker from './ScheduleMaker';
 import 'whatwg-fetch';
 import { shortenUrl } from '../constants/resources';
+import { addGrades } from '../actions/grades';
+import { getInstructorLastName, calculateGPA } from '../components/CourseTable';
 
-const generateSchedules = (values, loadQuery) => (dispatch, getState) => {
+const genSchedules = (courseList, gap, sort, gradeMap) => {
+  let schedules = [];
+  if (!courseList.find(list => list.length === 0)) {
+    const scheduleMaker = new ScheduleMaker(courseList, parseInt(gap, 10), sort);
+    try {
+      schedules = scheduleMaker.makeSchedules();
+      if (sort === 'yes') {
+        /** My Implementation of Insertion sort */
+        for (let x = 1; x < schedules.length; x += 1) {
+          const key = schedules[x];
```

Comment:
Why is this called `key`?

Reply:
The key holds the object in the array in which we try to 'insert' it into its correct location.
I don't know if you want a better name for it, but online resources seem to agree to call it a key.
Even the VT 2114 CS Textbook calls it a key.